source/particles/particle_handler.cc

```diff
@@ -378,7 +378,8 @@ namespace Particles
   {
     const unsigned int active_cell_index = particle->active_cell_index;
 
-    particle_container &container = *(particle->particles);
+    Assert(particle->particles == &particles, ExcInternalError());
+    particle_container &container = particles;
```

Comment:
This will probably only work after #12412 is merged. I am not sure if anyone is actually individually removing ghost_particles at the moment, but at least in principle they could.

Reply:
Oh, I see now -- the container could be the particles or ghost particles. I thought that this is actually a redundant specification. I'll close the PR then.